### PR TITLE
[image-builder] replace reference to aliases with actual repo for cross mount blobs

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Some registries like azure do not fail if the reference to the cross repo is incorrect while some fail due to it. This PR makes sure such references to aliases are appropriately replaced.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7264

## How to test

You can watch me blabbering how this works in this loom video or proceed to following sections to learn how to test.

[Cross mount repository fix for image-builder - Watch Video](https://www.loom.com/share/9603838832304816a177e6eb9c07c1f5)

Use the installer generated from this PR's build. You need a kubernetes cluster and need to install gitpod using the installer.

1. Create a gitlab registry
2. Create a personal access token with read and write permission to your registry
3. Create a docker secret using your credentials

```
kubectl create secret docker-registry registry-secret --docker-server=registry.gitlab.com --docker-username=prince41 --docker-password="mypassword"
```

4. Make the following changes to the gitpod config (generated via `./installer init > gitpod-config.yaml`)

```yaml
certificate:
  kind: secret
  name: https-certificates
containerRegistry:
  inCluster: false
  external:
    url: registry.gitlab.com/prince41/testproj
    certificate:
      kind: secret
      name: registry-secret
```

I also configured the experimental field as I tested this on workspace-preview which by default does an experimental installation:

```yaml
experimental:
  webapp:
    server:
      disableDynamicAuthProviderLogin: false
      enableLocalApp: true
      defaultBaseImageRegistryWhitelist:
      - https://index.docker.io/v1/
      - https://azurecr.io/
      - https://prince.azurecr.io/
      - https://prince.azurecr.io/base-images
      - https://prince.azurecr.io/workspace-images
      - https://registry.gitlab.io/
      - https://registry.gitlab.io/prince41/
      - https://registry.gitlab.io/prince41/testproj
```

Make sure you correctly configure `domain` before moving on to the next steps.

5. Generate gitpod yamls and apply

```console
installer render --use-experimental-config --config gitpod-config.yaml > rendered.yaml
kubectl apply -f rendered.yaml
```

Now use a repo which has a gitpod configuration similar to this:
```yaml
image:
  file: gitpod.Dockerfile
```

and the corresponding gitpod.Dockerfile:

```dockerfile
FROM gitpod/workspace-full:latest

RUN touch /tmp/abc && echo "done"
```

Image build should work without any issue.

image

## Release Notes

```release-note
Fix Cross mount blob error
```

### CODE REVIEW

File: components/image-builder-bob/pkg/proxy/proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
```
Comment (jenting) [inline components/image-builder-bob/pkg/proxy/proxy_test.go:1]: nit, it's 2022 now.

**Comment by sagor999** [comment]
@kylos101 I think previous production issue was due to pinning of bob to specific version. So we did test everything correctly on our end in that case.

**Comment by sagor999** [comment]
Rebased PR on latest main so that build will not fail.

**Comment by kylos101** [comment]
Thanks @sagor999 ! Have a nice vacation. :smiley: 

There are a few image builder PRs that we have out there, so I'll be sure to rebase them too. 

I plan to work with WebApp (@geropl ) on Tuesday to understand how related testing and deploys are done.  I am out on Monday.

I assume we first must remove a pin in the ops repo, but would also like how to deploy to staging, and eventually production for webapp. This should also help the team eventually get image-builder onto workspace clusters...which would be great!

**Comment by princerachit** [comment]
I have rebased and resolved conflict.

**Comment by kylos101** [comment]
Hi @sagor999 no, but, I've added as part of [this epic](https://github.com/gitpod-io/gitpod/issues/7845) (to step 1), which we'll be starting soon. :pray: Thank you for asking.

**Comment by sagor999** [comment]
Going to mark this review as draft, as I am not sure who is owning this PR at this moment. @kylos101

**Pull request closed without merging**